gnomad/utils/annotations.py

```diff
@@ -1293,107 +1286,106 @@ def get_gnomad_gks(
+        :param variant_ht: Hail Table with 1 row, only containing desired variant.
+        :param group_index: Index of frequency in variant_ht.freq containing the desired group.
+        :param group_id: String containing variant, pop, and sex. Example: chr19-41094895-C-T.afr.XX .
+        :param group_label: String containing the full name of pop requested. Example: African/African American.
+        :param vrs_id: String containing the VRS ID of the variant in ht_subpop.
         :return: Dictionary containing VRS information (and population if desired) for specified variant.
 
         """
 
-        # Frequency of desired variant to be parsed
-        variant_freq_to_parse = ht_subpop.freq[index_subpop]
+        # Obtain frequency information for the specified variant
+        group_freq = variant_ht.freq[group_index]
 
-        # Dictionary to be returned containing the subpop's information
+        # Dictionary to be returned containing information for a specified group
         subpop_record = {
             "subpopulationFrequency": [
                 {
-                    "id": id_subpop,
+                    "id": group_id,
                     "type": "PopulationAlleleFrequency",
-                    "label": f"{label_subpop} Population Allele Frequency for ID",
-                    "focusAllele": vrs_id_subpop,
-                    "focusAlleleCount": variant_freq_to_parse["AC"].collect()[0],
-                    "locusAlleleCount": variant_freq_to_parse["AN"].collect()[0],
-                    "alleleFrequency": variant_freq_to_parse["AF"].collect()[0],
-                    "population": f"gnomad3:{id_subpop}",
+                    "label": f"{group_label} Population Allele Frequency for ID",
+                    "focusAllele": vrs_id,
+                    "focusAlleleCount": group_freq["AC"].collect()[0],
+                    "locusAlleleCount": group_freq["AN"].collect()[0],
+                    "alleleFrequency": group_freq["AF"].collect()[0],
+                    "population": f"gnomad3:{group_id}",
                     "ancillaryResults": {
-                        "homozygotes": variant_freq_to_parse[
-                            "homozygote_count"
-                        ].collect()[0]
+                        "homozygotes": group_freq["homozygote_count"].collect()[0]
                     },
                 }
             ]
         }
 
         return subpop_record
 
-    # Iterate through provided ancestry groups and generate dictionaries
-    # If none are supplied, this is elegantly skipped!
-    for group in ancestry_groups:
-        key = f"{group}-adj"
-        index_value = ht.freq_index_dict.get(key)
-        result = create_subpops(
-            ht_subpop=ht,
-            index_subpop=index_value,
-            id_subpop=group,
-            label_subpop=POP_NAMES[group],
-            vrs_id_subpop=vrs_id,
-        )
+    # Iterate through provided groups and generate dictionaries
+    if groups:
```

Comment:
does this work for you? it looks like the dictionary keys are not unique so only one ancestry group gets returned

Reply:
it currently does work for me, likely because the current setup of the script passes in the keys to check as a list and then the dictionary separately